apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -225,11 +225,26 @@ struct SourceDirective {
 /// are resolvable by some combination of `@connect` directives.
 fn validate_object_fields(
     object: &Node<ObjectType>,
-    sources: &SourceMap,
+    schema: &Schema,
     connect_directive_name: &Name,
     source_directive_name: &Name,
     source_names: &[SourceName],
 ) -> Vec<Error> {
+    let source_map = &schema.sources;
+    let is_subscription = schema
+        .schema_definition
+        .subscription
+        .as_ref()
+        .is_some_and(|sub| sub.name == object.name);
+    if is_subscription {
+        return vec![Error {
+            code: ErrorCode::SubscriptionInConnectors,
+            message: "connectors can't be used to create subscriptions".to_string(),
```

Comment:
```suggestion
            message: "The `Subscription` root type is not supported when using `@{connect_directive_name}`".to_string(),
```
maybe?

Reply:
I was avoiding the `Subscription` name since it can be renamed, but just using the object type could also be confusing. I'll do something similar without the "proper" noun, though.